### ISSUE TITLE
Ansible 2.5.0 has a critical bug, please roll back to a prior version or make a tap of prior version available

### ISSUE THREAD

**Issue by jchonig** [Homebrew/homebrew-core#25840]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [X ] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [ ] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [X ] ran `brew update` and can still reproduce the problem?
- [ X] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [X ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [X ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)

Use Ansible targetting non-x86 non-bare metal host.

- What happened (include command output)

FAILED! => {"changed": false, "cmd": "None -s system-product-name", "msg": "[Errno 2] No such file or directory", "rc": 2}

- What you expected to happen

Error doesn't appear.

- Step-by-step reproduction instructions (by running `brew install` commands)

Run Ansible against a target that does not support dmidecode and it will fail.

Ansible 2.5.0 is broken unless your target is an x86 non-vm: https://github.com/ansible/ansible/issues/37850.  Please roll back to the previous version until this issue is fixed.

# Formula additions or changes
To get formulae added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
We will close issues requesting formulae changes.

**Comment by ilovezfs** [comment]
You should be able to add `gather_subset=!hardware` as a workaround.

**Comment by jchonig** [comment]
Yes, I can.  However I believe there are other things in 2.5.0 that are not
working.  Moving to a .0 release without having a rollback version is less
than ideal.

Thanks

Jeff

-- 
Jeffrey C. Honig <jch@honig.net>
http://jch.honig.net
GnuPG ID:14E29E13 <http://jch.honig.net/Home/pgp_key>
Keybase: jchonig <https://keybase.io/jchonig>

**Comment by ilovezfs** [comment]
Let us know if you can identify those other things and we'll take a look.

**Comment by core-process** [comment]
My temporary workaround (Ubuntu Artful, Ansible PPA):

```sh
cd
wget https://github.com/ansible/ansible/commit/0ad783c381fc811ca0b375ceba83eeab1914a08c.patch

cd /usr/lib/python2.7/dist-packages/
sudo patch -p2 -i ~/0ad783c381fc811ca0b375ceba83eeab1914a08c.patch

cd ansible/module_utils/facts/virtual
sudo python -m compileall linux.py

cd
rm 0ad783c381fc811ca0b375ceba83eeab1914a08c.patch
```

**Comment by core-process** [comment]
My temporary workaround for Ubuntu Artful using Ansible PPA. You should be able to adapt it to the Brew environment.

```sh
cd
wget https://github.com/ansible/ansible/commit/0ad783c381fc811ca0b375ceba83eeab1914a08c.patch

cd /usr/lib/python2.7/dist-packages/
sudo patch -p2 -i ~/0ad783c381fc811ca0b375ceba83eeab1914a08c.patch

cd ansible/module_utils/facts/virtual
sudo python -m compileall linux.py

cd
rm 0ad783c381fc811ca0b375ceba83eeab1914a08c.patch
```

**Comment by ilovezfs** [comment]
@jchonig this is now fixed in 2.5.1, which I've just shipped https://github.com/Homebrew/homebrew-core/pull/26800

**Comment by jchonig** [comment]
I'm now up to 2.5.2 and the workaround no longer seems to be required.

Thanks

Jeff

-- 
Jeffrey C. Honig <jch@honig.net>
http://jch.honig.net
GnuPG ID:14E29E13 <http://jch.honig.net/Home/pgp_key>
Keybase: jchonig <https://keybase.io/jchonig>
